lib/commands/element.ts

```diff
@@ -0,0 +1,150 @@
+  return (await this.uiautomator2.jwproxy.command(`/element/${elementId}/location`, 'GET', {})) as Position;
+}
+
+/**
+ * Gets the element size.
+ */
+export async function getSize(this: AndroidUiautomator2Driver, elementId: string): Promise<Size> {
+  return (await this.uiautomator2.jwproxy.command(`/element/${elementId}/size`, 'GET', {})) as Size;
+}
+
+/**
+ * Sets the value of an element using the upstream driver API.
+ */
+export async function doSetElementValue(this: AndroidUiautomator2Driver, params: DoSetElementValueOpts): Promise<void> {
+  await this.uiautomator2.jwproxy.command(`/element/${params.elementId}/value`, 'POST', params);
+}
+
+/**
+ * Sends text to an element without replacement.
+ */
+export async function setValueImmediate(
+  this: AndroidUiautomator2Driver,
+  keys: string | string[],
+  elementId: string,
+): Promise<void> {
+  await this.uiautomator2.jwproxy.command(`/element/${elementId}/value`, 'POST', {
+    elementId,
+    text: _.isArray(keys) ? keys.join('') : keys,
+    replace: false,
+  });
+}
+
+/**
+ * Gets the element text.
+ */
+export async function getText(this: AndroidUiautomator2Driver, elementId: string): Promise<string> {
+  return String(await this.uiautomator2.jwproxy.command(`/element/${elementId}/text`, 'GET', {}));
+}
+
+/**
+ * Clicks the given element.
+ */
+export async function click(this: AndroidUiautomator2Driver, element: string): Promise<void> {
+  await this.uiautomator2.jwproxy.command(`/element/${element}/click`, 'POST', {element});
+}
+
+/**
+ * Takes a screenshot of the element.
+ */
+export async function getElementScreenshot(this: AndroidUiautomator2Driver, element: string): Promise<string> {
+  return String(await this.uiautomator2.jwproxy.command(`/element/${element}/screenshot`, 'GET', {}));
+}
+
+/**
+ * Clears the element text.
+ */
+export async function clear(this: AndroidUiautomator2Driver, elementId: string): Promise<void> {
+  await this.uiautomator2.jwproxy.command(`/element/${elementId}/clear`, 'POST', {elementId});
+}
+
+/**
+ * Gets the element rectangle.
+ */
+export async function getElementRect(this: AndroidUiautomator2Driver, elementId: string): Promise<Rect> {
+  const chromedriver = this.chromedriver as any;
+  if (this.isWebContext()) {
+    this.log.debug(`Detected downstream chromedriver protocol: ${chromedriver.jwproxy.downstreamProtocol}`);
+    if (chromedriver.jwproxy.downstreamProtocol === PROTOCOLS.MJSONWP) {
+      const [{x, y}, {width, height}] = (await B.all([
+        chromedriver.jwproxy.command(`/element/${elementId}/location`, 'GET'),
+        chromedriver.jwproxy.command(`/element/${elementId}/size`, 'GET'),
+      ])) as [Position, Size];
+      return {x, y, width, height};
+    }
+    return (await chromedriver.jwproxy.command(`/element/${elementId}/rect`, 'GET')) as Rect;
+  }
+  return (await this.uiautomator2.jwproxy.command(`/element/${elementId}/rect`, 'GET')) as Rect;
```

Comment:
a small preference to return `!this.isWebContext()` first

Reply:
updated